File: src/main/java/org/taymyr/lagom/elasticsearch/client/ElasticSearch.java
package org.taymyr.lagom.elasticsearch.client;

import com.lightbend.lagom.javadsl.api.Descriptor;
import com.lightbend.lagom.javadsl.api.deser.PathParamSerializers;
import com.lightbend.lagom.javadsl.api.transport.Method;
import org.taymyr.lagom.elasticsearch.client.serialize.JsonBytes;
import org.taymyr.lagom.elasticsearch.client.serialize.JsonBytesSerializer;

import static com.lightbend.lagom.javadsl.api.Service.named;
import static com.lightbend.lagom.javadsl.api.Service.restCall;

public interface ElasticSearch extends ElasticSearchKt {

    @Override
    default Descriptor descriptor() {
        return named("elastic-search")
                .withCalls(
                        restCall(Method.GET, "/:index/:type/_search", this::search),
                        restCall(Method.POST, "/:index/:type/:id/_update", this::updateIndex),
                        restCall(Method.DELETE, "/:index/:type/:id", this::deleteById)
                ).withPathParamSerializer(String.class, PathParamSerializers.STRING)
                .withMessageSerializer(JsonBytes.class, new JsonBytesSerializer())
                .withAutoAcl(true);
    }

}

File: src/main/kotlin/org/taymyr/lagom/elasticsearch/client/ElasticSearch.kt
package org.taymyr.lagom.elasticsearch.client

import akka.Done
import akka.NotUsed
import com.lightbend.lagom.javadsl.api.Service
import com.lightbend.lagom.javadsl.api.ServiceCall
import org.taymyr.lagom.elasticsearch.client.dsl.QueryRoot
import org.taymyr.lagom.elasticsearch.client.serialize.JsonBytes

interface ElasticSearchKt : Service {
    fun updateIndex(index: String, type: String, id: String): ServiceCall<JsonBytes, Done>

    fun search(index: String, type: String): ServiceCall<QueryRoot, JsonBytes>

    fun deleteById(index: String, type: String, id: String): ServiceCall<NotUsed, Done>
}
